app/src/main/java/com/sebastianvm/mango/data/UserStore.kt

```diff
@@ -0,0 +1,36 @@
+package com.sebastianvm.mango.data
+
+import com.sebastianvm.mango.database.MangoDatabase
+import com.sebastianvm.mango.database.models.UserEnt
+import com.sebastianvm.mango.model.User
+import com.sebastianvm.mango.util.coroutines.IODispatcher
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ViewModelComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+
+class UserStore (
+    private val db: MangoDatabase,
+    private val ioDispatcher: CoroutineDispatcher
+) {
+
+    suspend fun getUserById(id: Int): User? = withContext(ioDispatcher) { db.userDao().getUserById(id) }
+
+}
+
+fun User.toEnt() = UserEnt(id, name)
+
+@Module
+@InstallIn(ViewModelComponent::class)
+object UserStoreModule {
+
+    @Provides
+    fun provideUserStore(
+        db: MangoDatabase,
+        @IODispatcher ioDispatcher: CoroutineDispatcher): UserStore {
+        return UserStore(db, ioDispatcher)
+    }
+
```

Comment:
Once we switch to interface, we can use `@Binds` to set up the module: https://developer.android.com/training/dependency-injection/hilt-android#inject-interfaces